R/remove_funs.R

```diff
@@ -22,12 +22,22 @@ crop <- function ( x , char = " " ) {
 
 ### gsub for more than one pattern
 gsubAll <- function(string, old, new) {
-  stopifnot ( is.character(string), is.character(old), is.character(new), length(old) == length(new))
+  checkmate::assert_character(string)
+  lapply(list(old, new), FUN = checkmate::assert_character,unique=TRUE, any.missing = FALSE, min.len = 2)
+  stopifnot ( length(old) == length(new))
+  stopifnot (length(intersect(old, new))==0)                                    ### in einem Zwischenschritt unique IDs mit gleich vielen characters erzeugen
+  inter <- stringi::stri_rand_strings(n=length(old), length=12, pattern="[A-Za-z0-9]")
+  stopifnot(length(unique(inter)) == length(inter))
```

Comment:
Is this `stopifnot()` line necessary? To my understanding, `length(inter)` is always `length(old)` because of the line above?

Reply:
Yes, `length(inter)` is always `length(old)`, but I was not sure whether `stringi::stri_rand_strings(...)` always produces unique codes. So the main question was (for me), whether `inter` is unique which is tested in line 30.